### PR TITLE
adding support for zero-scale

### PR DESCRIPTION
**Pull request by adinin**

### CODE REVIEW

File: autoscaler/template_processing.py [hunk 1]
```diff
@@ -110,6 +119,16 @@ def unroll_nic(template, pool, new_pool_size):
 
     return template
 
+def get_from_template(resource_name):
```
Comment (kevincvlam) [inline autoscaler/template_processing.py]: could you add a docstring explaining what this function does and why?

File: main.py [hunk 2]
```diff
@@ -99,6 +100,13 @@ def main(resource_group, acs_deployment, sleep, kubeconfig,
     cluster.login()
     backoff = sleep
     while True:
+        while True:
```
Comment (kevincvlam) [inline main.py]: for my understanding, what's going on in this loop? A comment may be helpful

File: autoscaler/kube.py [hunk 3]
```diff
@@ -67,8 +67,14 @@ def is_in_drain_grace_period(self):
         """
         return not self.start_time or (datetime.datetime.now(self.start_time.tzinfo) - self.start_time) < self._DRAIN_GRACE_PERIOD
 
+    def is_heapster(self):
+        """ determines whether the pod is a heapster pod
```
Comment (kevincvlam) [inline autoscaler/kube.py]: maybe explain what this is for and why we don't want to drain the heapster pod?

**Comment by cbockman** [comment]
Need type hints...

File: autoscaler/template_processing.py (hunk 1 above)
Comment (kevincvlam) [reply autoscaler/template_processing.py]: nit: should have Args: section

**Pull request merged**